Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractCompactRandomCutTree.java

```diff
@@ -150,9 +103,80 @@ protected void addSequenceIndex(Integer nodeRef, long sequenceIndex) {
         sequenceIndexes[leafRef].put(sequenceIndex, num + 1);
     }
 
+    public double getBoundingBoxCacheFraction() {
+        return boundingBoxCacheFraction;
+    }
+
+    abstract void swapCaches(int[] map);
+
+    public void renormalize() {
+        NodeStore result = new NodeStore(maxSize - 1);
+        if (root != null) {
+            if (!isLeaf(root)) {
+                int nodeCounter = 0;
+                int currentNode = 0;
+                int leafcounter = maxSize - 1;
+                int[] map = new int[2 * maxSize - 1];
+                ArrayBlockingQueue<Integer> nodeQueue = new ArrayBlockingQueue<>(maxSize - 1);
```

Comment:
Do we really need a blocking queue? If not, we could just use a LInkedList here.

Reply:
We need the strong FIFO property -- but we can use a linked list.